xrootd/client/filesystem.go

```diff
@@ -31,6 +32,16 @@ func (fs *fileSystem) Dirlist(ctx context.Context, path string) ([]xrdfs.EntrySt
 	return resp.Entries, err
 }
 
+// Open returns the file handle for a file together with the compression and the stat info.
+func (fs *fileSystem) Open(ctx context.Context, path string, mode xrdfs.OpenMode, options xrdfs.OpenOptions) (xrdfs.File, error) {
+	var resp open.Response
+	err := fs.c.Send(ctx, &resp, open.NewRequest(path, mode, options))
+	if err != nil {
+		return file{}, err
+	}
+	return file{fs, resp.FileHandle, resp.Compression, resp.Stat}, nil
```

Comment:
you probably want to `return &file{...}` here (and `return nil, err` above).
right now, both `client.file` and `*client.file` implement `xrdfs.File` but I suspect that later on, only `*client.file` will be able to implement it (_ie:_  for all the methods that modify the internal state of `client.file`, a pointer-receiver will be required.)